core/src/main/java/apoc/refactor/GraphRefactoring.java

```diff
@@ -452,6 +458,76 @@ public void categorize(
+    public Stream<GraphResult> deleteAndReconnect(@Name("path") Path path, @Name("nodes") List<Node> nodesToRemove, @Name(value = "config", defaultValue = "{}") Map<String,Object> config) {
+
+        RefactorConfig refactorConfig = new RefactorConfig(config);
+
+        List<Node> nodes = IterableUtils.toList(path.nodes());
+        Set<Relationship> rels = Iterables.asSet(path.relationships());
+
+        if (!nodes.containsAll(nodesToRemove)) {
+            throw new RuntimeException("Some nodes in nodesToRemove not present in the path");
+        }
+
+        BiFunction<Node, Direction, Relationship> filterRel = (node, direction) -> StreamSupport
+                .stream(node.getRelationships(direction).spliterator(), false)
+                .filter(rels::contains)
+                .findFirst()
+                .orElse(null);
+
+        nodesToRemove.forEach(node -> {
+
+            Relationship relationshipIn = filterRel.apply(node, Direction.INCOMING);
+            Relationship relationshipOut = filterRel.apply(node, Direction.OUTGOING);
+
+            // if terminal node
+            if (relationshipIn == null || relationshipOut == null) {
+                rels.remove(relationshipIn == null ? relationshipOut : relationshipIn);
+
+            } else {
+                Node nodeIncoming = relationshipIn.getStartNode();
+                Node nodeOutcoming = relationshipOut.getEndNode();
+
+                RelationshipType newRelType;
+                Map<String, Object> newRelProps = new HashMap<>();
+
+                final RefactorConfig.RelationshipSelectionStrategy strategy = refactorConfig.getRelationshipSelectionStrategy();
+                switch (strategy) {
+                    case INCOMING:
+                        newRelType = relationshipIn.getType();
+                        newRelProps.putAll(relationshipIn.getAllProperties());
+                        break;
+
+                    case OUTCOMING:
+                        newRelType = relationshipOut.getType();
+                        newRelProps.putAll(relationshipOut.getAllProperties());
+                        break;
+
+                    default:
+                        newRelType = RelationshipType.withName(relationshipIn.getType() + "_" + relationshipOut.getType());
+                        newRelProps.putAll(relationshipIn.getAllProperties());
+                }
+
+                Relationship relCreated = nodeIncoming.createRelationshipTo(nodeOutcoming, newRelType);
+                newRelProps.forEach(relCreated::setProperty);
+
+                if (strategy == MERGE) {
+                    mergeProperties(relationshipOut.getAllProperties(), relCreated, refactorConfig);
+                }
+
+                rels.add(relCreated);
+                rels.removeAll(List.of(relationshipIn, relationshipOut));
+            }
+
+            tx.execute("WITH $node as n DETACH DELETE n", Map.of("node", node));
```

Comment:
where is the Cypher query that connects the previous and next nodes together?

Reply:
On line 513, `Relationship relCreated = nodeIncoming.createRelationshipTo(nodeOutcoming, newRelType);`
This line deletes the intermediate node.
So with a path like `(A)-->(B)-->(C)`, if i want to delete node `B`, first I create connection between `A` and `C`, and then I delete `B`.